### ISSUE TITLE
Expected runtime

### ISSUE THREAD

**Issue by willishf** [NVIDIA/DIGITS#1515]
Working on a tumor tissue classification competition for https://camelyon17.grand-challenge.org/ where submission deadline is April 1st. Spent a substantial amount of time annotating the supplied training images for the different visually distinct regions in the tumor. I then have code that will automatically export 256x256 images from the annotated regions and rotate/jitter multiple versions of the image. Cell shape/color/spacing are distinct for the different classes of images. Overall have about 2.5 million images where each image is rotated 10 degrees and random jitter introduced. Each image at 256x256 usually contains 8 x 8 cells of similar patterns/colors where each label will have distinct cell features related to color and size. 

I am in the process of creating the database as the first step using DIGITS 5.0 web interface and after running for an hour the estimated time remaining for the train DB is 3 days and validation is 1 day. Running on a GTX 1080 and a 8 core processor with 16 GB of ram. CPU load is around 30 percent and memory usage a 35%. Unfortunately the images are stored on a 6TB external USB hard drive formatted as NTFS. Shuttling the HD from a windows computer to do the annotations to ubuntu 16 box running digits. I suspect the HD for the source of images is the bottleneck. Database is being stored on internal HD.  

I was looking for some guidance from power users that have worked with large data sets. 

1. If it is taking 3 days just to create the database I assume the creating of the model will take much longer? Any guidance on the time it takes to create the DB as an indicator of time to create a Image classification model with the defaults? I can easily reduce the number of images and reduce the number of rotated images. 

2. When creating the model with 2 million images is the default number of 30 epochs and base learning rate of 0.01 appropriate? Trying to avoid multiple iterations of making models if it will be an extended amount of time per model. 

3. Any recommendations on solver type?

4. Should I make any changes to batch size?

5. Any recommendations on AlexNet vs GoogLeNet?

Some example images to help provide some context on the features/recognition problem for training as guidance for the expected number of training images. Of course many variations on these images. A team from google is also participating in the competition and recently published the approach they took and the number of images were 10,000,000. Not clear if all 10 million were used since they were sampled based on tissue type and no indication of the number of tissue types. 

https://drive.google.com/file/d/0B1T58bZ5vYa-QlR0QlJTa2dPWVk/view

Any guidance would be appreciated. 

Tumor

![image](https://cloud.githubusercontent.com/assets/4054430/24002955/76d64db2-0a38-11e7-9fa8-373a17be7d0a.png)

Normal
![image](https://cloud.githubusercontent.com/assets/4054430/24003006/9ac07a86-0a38-11e7-8e29-df501b3773b8.png)

Fat
![image](https://cloud.githubusercontent.com/assets/4054430/24003037/b2156386-0a38-11e7-849d-d0daaa904a54.png)

Red Blood Cell
![image](https://cloud.githubusercontent.com/assets/4054430/24003077/d7d48c3c-0a38-11e7-9ae6-c0e650360cbf.png)

Capsule

![image](https://cloud.githubusercontent.com/assets/4054430/24003108/f5c6bc74-0a38-11e7-994d-97398f540d61.png)

**Comment by gheinrich** [comment]
Hello, it sounds like you'd be better off doing online (i.e. during training) augmentation. This would reduce the size of your dataset tremendously. Common augmentation tricks are implemented for you in DIGITS if you use the Torch backend.

**Comment by willishf** [comment]
Can you provide details/tutorial on online augmentation? I assume this means the images will be transformed/rotated and thus no reason for me to provide rotated images?

I have torch installed but haven't tested. Is that a selection via the digits web interface?

Is my assumption correct that creating the database is generally much faster then generating the model?

**Comment by gheinrich** [comment]
Hello, see this doc:
https://github.com/NVIDIA/DIGITS/blob/digits-5.0/docs/GettingStartedTorch.md
You will find the augmentation options on the model page.

> Is my assumption correct that creating the database is generally much faster then generating the model?

Generally yes.

**Comment by willishf** [comment]
Definitely learned the hard way that USB 3.0 and external hard drive is not an optimal medium for a couple million small images 256x256 to be used in training. Copied everything over to SSD SATA drives. Database creation from SSD SATA took 6 hours as opposed to 3 days on USB 3.0 drive.

Went through training for image classification of 2.7 million 256x256 images in six classes use caffe and googlenet. After 24 hours completed 6 epochs where after 1st epoch 99% accuracy for validation images. I have a couple classes that represent transition from one image type to another that don't have the same number of representative images.

Preliminary testing of images not used in training and validation give 99% on the correct image type. For the classes that represent a transition from one class to another(50% of one image time and 50% of another) the prediction is not very good but picking an appropriate choice in that I wouldn't be able to tell the difference in the context of the single image. 

Currently have completed 8 epochs and don't see how it can improve based on validation and training metrics. Does it make sense to let it keep running on the models with the hopes of improving the classes/labels for transition from one image type to another(I provide those images as a separate label)? Looking for guidance on the expectation that letting it keep running would expect any enhancements if already at 99% for validation and testing images types that matter return a confidence of 99%. Other top 5 labels are below 1%.

I would export the models for the last 7 epochs and stop processing so I can move on to the classifying of test images.  

![image](https://cloud.githubusercontent.com/assets/4054430/24108323/a7c6b69a-0d63-11e7-8752-d651d68d8634.png)


![image](https://cloud.githubusercontent.com/assets/4054430/24108285/8f4ff126-0d63-11e7-8a67-89397e79eba3.png)
